src/oemof/solph/components/_generic_storage.py

```diff
@@ -207,20 +208,34 @@ def __init__(
                 + " nominal_storage_capacity."
                 + " Both options cannot be set at the same time."
             )
-            if nominal_storage_capacity is not None:
+            if nominal_capacity is not None:
                 raise AttributeError(msg)
             else:
                 warn(msg, FutureWarning)
             nominal_storage_capacity = investment
         # --- END ---
+        # --- BEGIN: The following code can be removed for versions >= v0.6 ---
+        if nominal_storage_capacity is not None:
+            msg = (
+                "For backward compatibility,"
+                " the option nominal_storage_capacity overwrites the option"
+                + " nominal_capacity."
+                + " Both options cannot be set at the same time."
```

Comment:
Minor remark: The usage of string concatenation is inconsistent here. I'd suggest to remove the pluses as they are not necessarily needed.

Reply:
The plus helps to see that the line continues the last one. Thus, I added the missing one.